common/persistence/historyStore.go

```diff
@@ -445,7 +445,7 @@ func (m *historyV2ManagerImpl) readHistoryBranch(
 			}
 		}
 
-		token.LastEventVersion = firstEvent.GetVersion()
```

Comment:
Was there a real bug or this is just for consistency? 
Version on first and last event should be the same.

If this fixes a real bug then you should consider adding a unit test for it.

Reply:
consistency, the above code already assert version being the same